jpa/eclipselink.jpa.test.jse/src/it/java/org/eclipse/persistence/jpa/json/TestJson.java

```diff
@@ -254,4 +255,32 @@ public void testSelectJsonInWhereCondition() {
         }
     }
 
+    @Test
+    public void testEscapedQuestionMarkInSQLOperator() {
+        EntityManager em = emf.createEntityManager();
+
+        if (emf.unwrap(Session.class).getPlatform().isOracle()) {
+            JsonValue value = Json.createObjectBuilder()
+                    .add("id", "1007")
+                    .build();
+            try {
+                em.getTransaction().begin();
+                JsonEntity e = new JsonEntity(1007, value);
+                em.persist(e);
+                em.flush();
+                em.getTransaction().commit();
+                em.clear();
+
+                JsonEntity dbValue = em.createQuery(
+                    "SELECT v.value FROM JsonEntity v WHERE SQL('JSON_EXISTS(?, ''$??(@.id == 1007)'')', v.value)", JsonEntity.class)
```

Comment:
I think, that correct query should be `SELECT v FROM JsonEntity.....` not `SELECT v.value FROM JsonEntity.....`.